### ISSUE TITLE
Timeline support request

### ISSUE THREAD

**Issue by marcusx2** [playcanvas/editor#462]
Please see [this video](https://www.youtube.com/watch?v=EqY-0jQH3aE). I think it would be nice if PlayCanvas had a similar feature. And being able to choose an interpolation method between keyframes like the video shows. Download [Studio](https://zap.works/studio/) and play a little with their timeline feature. Being able to make a bezier path would also be nice to move a model on a curve.

**Comment by yaustar** [comment]
This is currently on the roadmap as 'Animation dope sheet editor'. In the short term, some simple animations can be done via the tween library: https://github.com/playcanvas/playcanvas-tween

**Comment by marcusx2** [comment]
@yaustar Does it include bezier path to move objects on a curve?

**Comment by yaustar** [comment]
We haven't specc'ed it yet or started work on it yet

**Comment by yaustar** [comment]
Are you asking about playcanvas-tween? If so, no it doesn't. It just supports simple tweens.

**Comment by marcusx2** [comment]
No, not playcanvas-tween. The animation dope sheet editor. It would be really great if it supported bezier curves for curved path animations.

**Comment by yaustar** [comment]
The animation dope sheet editor hasn't been spec'ed or started work yet.

**Comment by marcusx2** [comment]
Oh I see...please consider my request then. Thanks!

**Comment by yaustar** [comment]
Using https://www.theatrejs.com/ could be a consideration here and integrating the runtime and editor to our own.

LeXXik has made a runtime version of this here: https://forum.playcanvas.com/t/playcanvas-theatre-js/25376

**Comment by marcusx2** [comment]
I approve the idea. Theatrejs already has all the logic we need for this feature request, and it's very polished. The effort of integrating theatrejs with PlayCanvas might be considerably less than developing this feature request from scratch.

**Comment by TheShoes** [comment]
+1  
this would be an amazing tool that I could see us using instantly.

**Comment by allancozart** [comment]
+1
Agreed, this would be a game changer for my team and would get us to switch from using other web based 3D engines.

**Comment by ideathingdev** [comment]
Yes please, this feature would have made my work so far on playcanvas so much easier and better. 
Hoping to see it coming soon
